vendor/openfx/OCIOUtils.cpp

```diff
@@ -0,0 +1,651 @@
+    OFX::ClipDescriptor * dstClip = desc.defineClip(
+        kOfxImageEffectOutputClipName);
+
+    dstClip->addSupportedComponent(OFX::ePixelComponentRGBA);
+    dstClip->addSupportedComponent(OFX::ePixelComponentRGB);
+}
+
+OCIO::ConstConfigRcPtr getOCIOConfig()
+{
+    OCIO::ConstConfigRcPtr config = OCIO::GetCurrentConfig();
+    if (!config)
+    {
+        config = OCIO::Config::CreateFromEnv();
+        OCIO::SetCurrentConfig(config);
+    }
+
+    return config;
+}
+
+OCIO::BitDepth getOCIOBitDepth(OFX::BitDepthEnum ofxBitDepth)
+{
+    OCIO::BitDepth ocioBitDepth = OCIO::BIT_DEPTH_UNKNOWN;
+
+    switch (ofxBitDepth)
+    {
+        case OFX::eBitDepthUByte:
+            ocioBitDepth = OCIO::BIT_DEPTH_UINT8;
+            break;
+        case OFX::eBitDepthUShort:
+            ocioBitDepth = OCIO::BIT_DEPTH_UINT16;               
+            break;
+        case OFX::eBitDepthHalf:
+            ocioBitDepth = OCIO::BIT_DEPTH_F16;               
+            break;
+        case OFX::eBitDepthFloat:
+            ocioBitDepth = OCIO::BIT_DEPTH_F32;
+            break;
+        case OFX::eBitDepthNone:
+        case OFX::eBitDepthCustom:
+        default:
+            std::ostringstream os;
+            os << "Unsupported bit-depth: ";
+            os << OFX::mapBitDepthEnumToStr(ofxBitDepth);
+            OFX::Log::error(true, os.str().c_str());
+            OFX::throwSuiteStatusException(kOfxStatErrUnsupported);
+    }
+
+    return ocioBitDepth;
+}
+
+int getChanStrideBytes(OCIO::BitDepth ocioBitDepth)
+{
+    int chanStrideBytes = 0;
+
+    switch (ocioBitDepth)
+    {
+        case OCIO::BIT_DEPTH_UINT8:
+            chanStrideBytes = 1;
+            break;
+        case OCIO::BIT_DEPTH_UINT16:
+        case OCIO::BIT_DEPTH_F16:
+            chanStrideBytes = 2;               
+            break;
+        case OCIO::BIT_DEPTH_F32:
+            chanStrideBytes = 4;
+            break;
+        case OCIO::BIT_DEPTH_UNKNOWN:
+        case OCIO::BIT_DEPTH_UINT10:
+        case OCIO::BIT_DEPTH_UINT12:
+        case OCIO::BIT_DEPTH_UINT14:
+        case OCIO::BIT_DEPTH_UINT32:
+        default:
+            std::ostringstream os;
+            os << "Unsupported bit-depth: ";
+            os << OCIO::BitDepthToString(ocioBitDepth);
+            OFX::Log::error(true, os.str().c_str());
+            OFX::throwSuiteStatusException(kOfxStatErrUnsupported);
+    }
+
+    return chanStrideBytes;
+}
+
+void defineCsNameParam(OFX::ImageEffectDescriptor & desc,
+                       OFX::PageParamDescriptor * page,
+                       const std::string & name, 
+                       const std::string & label, 
+                       const std::string & hint,
+                       OFX::GroupParamDescriptor * parent)
+{
+    OFX::ChoiceParamDescriptor * param = desc.defineChoiceParam(name);
+    initParam(param, name, label, hint, parent);
+
+    OCIO::ConstConfigRcPtr config = getOCIOConfig();
+
+    // Populate color space names
```

Comment:
To the extent that these plug-ins will serve as examples to app developers, it would be nice to encourage people to use ColorSpaceMenuHelper.  Perhaps add something like: "TODO: Use ColorSpaceMenuHelper to generate the menus in order to leverage features such as categories."

Reply:
Added.